pkg/filters/data.go

```diff
@@ -157,74 +324,162 @@ func (af *DataFilter) Parse(filterName string, operatorAndValues string, eventsN
 		return errfmt.WrapError(err)
 	}
 
-	af.Enable()
+	df.Enable()
 
 	return nil
 }
 
 // parseFilter adds an data filter with the relevant filterConstructor
 // The user must responsibly supply a reliable Filter object.
-func (af *DataFilter) parseFilter(id events.ID, dataName string, operatorAndValues string, filterConstructor func() Filter[*StringFilter]) error {
-	if _, ok := af.filters[id]; !ok {
-		af.filters[id] = map[string]Filter[*StringFilter]{}
+func (df *DataFilter) parseFilter(id events.ID, dataName string, operatorAndValues string, filterConstructor func() Filter[*StringFilter]) error {
+	if _, ok := df.filters[id]; !ok {
+		df.filters[id] = map[string]Filter[*StringFilter]{}
 	}
 
-	if _, ok := af.filters[id][dataName]; !ok {
+	if _, ok := df.filters[id][dataName]; !ok {
 		// store new event data filter if missing
 		dataFilter := filterConstructor()
-		af.filters[id][dataName] = dataFilter
+		df.filters[id][dataName] = dataFilter
 	}
 
 	// extract the data filter and parse expression into it
-	f := af.filters[id][dataName]
+	f := df.filters[id][dataName]
 	err := f.Parse(operatorAndValues)
 	if err != nil {
 		return errfmt.WrapError(err)
 	}
 
 	// store the data filter again
-	af.filters[id][dataName] = f
+	df.filters[id][dataName] = f
 
 	return nil
 }
 
-func (af *DataFilter) Enable() {
-	af.enabled = true
-	for _, filterMap := range af.filters {
+func (df *DataFilter) Enable() {
+	df.enabled = true
+	for _, filterMap := range df.filters {
 		for _, f := range filterMap {
 			f.Enable()
 		}
 	}
 }
 
-func (af *DataFilter) Disable() {
-	af.enabled = false
-	for _, filterMap := range af.filters {
+func (df *DataFilter) processKernelFilter(id events.ID, val, dataName string) (string, error) {
+	// Check for kernel filter restrictions
+	if err := df.checkKernelFilterRestrictions(val); err != nil {
+		return val, err
+	}
+
+	// Enable the kernel filter if restrictions are satisfied
+	df.enableKernelFilterArg(id, dataName)
+	return val, nil
+}
+
+// CheckKernelFilterRestrictions enforces restrictions for kernel-filtered
+// fields: 1) values cannot use "contains" (e.g., start and end with "*");
+// and 2) Maximum length for the value is 255 characters.
+func (df *DataFilter) checkKernelFilterRestrictions(val string) error {
+	if len(val) == 0 {
+		return InvalidValue("empty value is not allowed")
+	}
+
+	// Disallow "*" and "**" as invalid values
+	if val == "*" || val == "**" {
+		return InvalidValue(val)
+	}
+
+	// Check for "contains" type filtering
+	if len(val) > 1 && val[0] == '*' && val[len(val)-1] == '*' {
+		return InvalidFilterType()
+	}
```

Comment:
Using `strings.Count()` will consider cases as `"*some*hing"` invalid as well.

Would this suggestion work as expected?

```suggestion
	// Disallow a single "*" as invalid value
	if val == "*" {
		return InvalidValue(val)
	}

	// Check for "contains" type filtering
	if len(val) > 1 && strings.Count(val, "*") > 1 {
		return InvalidFilterType()
	}
```

Reply:
That makes sense, thank you for highlighting this. Today if we define a filter like this`/tmp/file\*`, this will be added as prefix (maybe the correct should be exact). In the end both ways will match a file access since `/tmp/file*` will be matched by the prefix `/tmp/file`.